File: backend/app/schemas/audit.py
from datetime import datetime

from pydantic import BaseModel


class AuditLogResponse(BaseModel):
    id: int
    timestamp: datetime
    user_id: int | None
    action: str
    target_type: str | None
    target_id: int | None
    details: str | None
    ip_address: str | None

    model_config = {"from_attributes": True}

File: frontend/src/api/audit.ts
import client from "./client";

export interface AuditLog {
  id: number;
  timestamp: string;
  user_id: number | null;
  action: string;
  target_type: string | null;
  target_id: number | null;
  details: string | null;
  ip_address: string | null;
}

export const auditApi = {
  list: (params?: { limit?: number; offset?: number; target_type?: string }) =>
    client.get<AuditLog[]>("/audit", { params }),
};
